core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -393,13 +393,17 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                                     .map(NodeRef::of)
                                     .map(columnReferences::get)
                                     .filter(Objects::nonNull)
+                                    .map(ResolvedField::getFieldId)
                                     .collect(toImmutableList());
                             for (Expression sortKey : sortKeys) {
-                                if (!node.getArguments().contains(sortKey) && !fieldIds.contains(columnReferences.get(NodeRef.of(sortKey)))) {
-                                    throw semanticException(
-                                            EXPRESSION_NOT_IN_DISTINCT,
-                                            sortKey,
-                                            "For aggregate function with DISTINCT, ORDER BY expressions must appear in arguments");
+                                if (!node.getArguments().contains(sortKey)) {
+                                    ResolvedField field = columnReferences.get(NodeRef.of(sortKey));
+                                    if ((field == null || !fieldIds.contains(field.getFieldId()))) {
```

Comment:
redundant braces?

Reply:
idk, but note that not every brace redundant from JLS perspective is redundant from human perspective too